src/plugins/dev_tools/public/application.tsx

```diff
@@ -91,7 +91,13 @@ function DevToolsWrapper({
             if (mountedTool.current) {
               mountedTool.current.unmountHandler();
             }
-            const params = { element, appBasePath: '', onAppLeave: () => undefined };
+            const params = {
+              element,
+              appBasePath: '',
+              onAppLeave: () => undefined,
+              // TODO: adapt to use Core's ScopedHistory
+              history: {} as any,
```

Comment:
It looks like the `ScopedHistory` class is exposed by core, will be as simple as passing down the history object from the application service and calling `new ScopedHistory(applicationServiceHistory, getDevToolBasePath(activeDevTool.id))`?

Reply:
Actually you'll want to use the `history.createSubHistory(subpathForDevTool)`. It's exported from core for the typing purposes only (once we have TS 3.8, this will be changed to a type-only export).